Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -41,90 +84,122 @@ extension ResourceOptions {
-                            baseUrl: String? = nil,
-                            cachePath: String? = nil,
-                            assetPath: String? = Bundle.main.resourceURL?.path,
-                            cacheSize: UInt64 = (1024*1024*50),
-                            tileStore: TileStore? = nil,
-                            tileStoreUsageMode: TileStoreUsageMode = .readOnly) {
-
-        // Update the TileStore with the access token from the ResourceOptions
-        if tileStoreUsageMode != .disabled {
-            let tileStore = tileStore ?? TileStore.default
-            tileStore.setAccessToken(accessToken)
-        }
-
-        let cacheURL = try? ResourceOptions.cacheURLIncludingSubdirectory()
-        let resolvedCachePath = cachePath == nil ? cacheURL?.path : cachePath
-        self.init(
-            __accessToken: accessToken,
-            baseURL: baseUrl,
-            cachePath: resolvedCachePath,
-            assetPath: assetPath,
-            cacheSize: NSNumber(value: cacheSize),
-            tileStore: tileStore,
-            tileStoreUsageMode: tileStoreUsageMode
-        )
+    public init(accessToken: String,
+                baseURL: URL? = nil,
+                cachePathURL: URL? = nil,
+                assetPathURL: URL? = nil,
+                cacheSize: UInt64 = Self.defaultCacheSize,
+                tileStore: TileStore? = nil,
+                tileStoreUsageMode: TileStoreUsageMode = .readOnly) {
+        self.accessToken        = accessToken
+        self.baseURL            = baseURL
+        self.cachePathURL       = cachePathURL ?? ResourceOptions.cacheURLIncludingSubdirectory()
+        self.assetPathURL       = assetPathURL ?? Bundle.main.resourceURL
+        self.cacheSize          = cacheSize
+        self.tileStore          = tileStore
+        self.tileStoreUsageMode = tileStoreUsageMode
     }
 
-    /// The size of the cache in bytes
-    public var cacheSize: UInt64? {
-        __cacheSize?.uint64Value
-    }
+    private static func cacheURLIncludingSubdirectory() -> URL {
+        do {
+            var cacheDirectoryURL = try FileManager.default.url(for: .applicationSupportDirectory,
+                                                                in: .userDomainMask,
+                                                                appropriateFor: nil,
+                                                                create: true)
 
-    private static func cacheURLIncludingSubdirectory() throws -> URL? {
-        var cacheDirectoryURL = try FileManager.default.url(for: .applicationSupportDirectory,
-                                                            in: .userDomainMask,
-                                                            appropriateFor: nil,
-                                                            create: true)
+            cacheDirectoryURL = cacheDirectoryURL.appendingPathComponent(".mapbox")
```

Comment:
so you make this change in the same PR? probably need to include this change to the change log.

Reply:
Yep good call. Adding here for convenience :)